elyra/pipeline/component_catalog.py

```diff
@@ -44,40 +50,168 @@
+
+                    pending_actions = manifest
+                    # If the 'source' is not the same as the manifest filename for this process,
+                    # the cache action has been triggered by a different process
+                    if source != self._manifest_filename and os.path.isfile(source):
+                        # Load out-of-process manifest in order to perform cache actions in-process
+                        with open(source, 'r') as f:
+                            oop_manifest = json.load(f)
+                        pending_actions = oop_manifest
+
+                    # Handle all pending actions in one batch (last update wins)
+                    for catalog_name, cache_action in pending_actions.items():
+                        if cache_action == 'delete':
+                            # Fabricate a metadata instance that only includes catalog name
+                            catalog_instance = ComponentCatalogMetadata(name=catalog_name)
+
+                        else:  # cache_action == 'modify':
+                            # Fetch the catalog instance associated with this action
+                            catalog_instance = MetadataManager(
+                                schemaspace=ComponentCatalogs.COMPONENT_CATALOGS_SCHEMASPACE_ID
+                            ).get(name=catalog_name)
+
+                        # Add action to the cache queue
+                        self._cache_queue.put((catalog_instance, cache_action))
+
+                        # Reset manifest actions to prepare for write out if this cache update
+                        # is acting upon its own manifest file
+                        if source == self._manifest_filename:
+                            self.build_manifest_file()
+
+                        self.manifest_update_pending = False
+
+                else:
+                    # Any action that is not a 'delete-manifest' or 'cache' action means
+                    # that an update to the manifest is required.
+                    catalog_name = source
+                    if manifest.get(catalog_name) != action:
+                        # Add new action to manifest if not already present
+                        manifest[catalog_name] = action
+
+                        # Write to file
+                        with open(self._manifest_filename, 'w') as f:
+                            json.dump(manifest, f, indent=2)
+
+                        self.manifest_update_pending = True
+
+                self._manifest_queue.task_done()
+
+            self.manage_cache_tasks()
+
+    def build_manifest_file(self):
+        """
+        Build an empty manifest and overwrite or create the file. Only the
+        CacheUpdateManagerThread is able to touch this file.
+        """
+        empty_manifest: Dict[str, str] = {}
+
+        with open(self._manifest_filename, 'w') as f:
+            f.write(json.dumps(empty_manifest, indent=2))
```

Comment:
Let's convert this to:
```suggestion
            json.dump(empty_manifest, f, indent=2)
```